termuic/termuiConsoleGrid.go

```diff
@@ -0,0 +1,162 @@
+package termuic
+
+import (
+	"fmt"
+	"strconv"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+type termuiConsoleGrid struct {
+	grid       *ui.Grid
+	lLog       *widgets.List
+	pPublicKey *widgets.Paragraph
+	pShardId   *widgets.Paragraph
+
+	pCountLeader         *widgets.Paragraph
+	pCountConsensus      *widgets.Paragraph
+	pCountAcceptedBlocks *widgets.Paragraph
+
+	gTxPoolLoad        *widgets.Gauge
+	pNumConnectedPeers *widgets.Paragraph
+
+	tSyncInfo *widgets.Table
+}
+
+//NewtermuiConsoleGrid initialize struct termuiConsoleGrid
+func NewtermuiConsoleGrid() *termuiConsoleGrid {
+
+	self := termuiConsoleGrid{}
+
+	self.initWidgets()
+
+	self.setupGrid()
+
+	return &self
+}
+
+func (tcg *termuiConsoleGrid) initWidgets() {
+
+	tcg.pPublicKey = widgets.NewParagraph()
+	tcg.pShardId = widgets.NewParagraph()
+
+	tcg.pCountLeader = widgets.NewParagraph()
+	tcg.pCountConsensus = widgets.NewParagraph()
+	tcg.pCountAcceptedBlocks = widgets.NewParagraph()
+
+	tcg.gTxPoolLoad = widgets.NewGauge()
+	tcg.pNumConnectedPeers = widgets.NewParagraph()
+
+	tcg.lLog = widgets.NewList()
+
+	tcg.tSyncInfo = widgets.NewTable()
+}
+
+func (tcg *termuiConsoleGrid) setupGrid() {
+	tcg.grid = ui.NewGrid()
+
+	tcg.grid.Set(
+		ui.NewRow(1.0/7, tcg.pPublicKey),
+		ui.NewRow(1.0/10, tcg.tSyncInfo),
+		ui.NewRow(1.0/3,
+			ui.NewCol(1.0/2,
+				ui.NewRow(1.0/4, tcg.pShardId),
+				ui.NewRow(1.0/4, tcg.pCountConsensus),
+				ui.NewRow(1.0/4, tcg.pCountLeader),
+				ui.NewRow(1.0/4, tcg.pCountAcceptedBlocks),
+			),
+			ui.NewCol(1.0/2,
+				ui.NewRow(1.0/2, tcg.gTxPoolLoad),
+				ui.NewRow(1.0/2, tcg.pNumConnectedPeers),
+			),
+		),
+		//ui.NewRow(1.0/3, tcg.slgNode),
+		ui.NewRow(1.0/3, tcg.lLog),
+	)
+}
+
+func (tcg *termuiConsoleGrid) SetupBigLogGrid() {
+	tcg.grid = ui.NewGrid()
+
+	tcg.grid.Set(
+		ui.NewRow(1.0, tcg.lLog),
```

Comment:
magic number, please extract this to a new variable

Reply:
There a lot of magic numbers because in this section of code needs to specify the number of column for every row  or de number of rows for every column, so for the moment for first version there is some magic numbers, if a change the view of console again the magic numbers need to be change.

Reply:
done